### PR TITLE
docs(kit): updated ecopasskit software development view, added new diagrams and adjusted content to reference implementation

### PR DESCRIPTION
**Pull request by matbmoser**
## Why do we create this PR:
There is the requirement to update the Eco Pass KIT development view

## What do we what to do in this PR
We want to update the development view, and add the necessary diagrams, content and descriptions to match the reference implementation of the  Digital Product Pass.

### Added
- Added new sequence diagrams in software architecture
- Added detailed data retrieval flow

### Updated
- Updated Software Architecture guide with descriptive documentation 

### Issues Fixed
- Corrected links paths to absolute path

## Content Review PRs:
- https://github.com/eclipse-tractusx/eco-pass-kit/pull/29

## Related PRs:
- [**Adoption View**] -> #630 

This is the second PR from the EcoPass KIT v2.0.0 Update

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by matbmoser** [comment]
the file `dataRetrievalFlow.jpg` was given to me in PowerPoint format, I have not the capacity now to translate it to Draw.io or something else. The other diagrams I added as .svg so they can be editable.

**Comment by matbmoser** [comment]
I am the author of all the images, even if it says that is 47k code changes, no IP check is necessary since I was the one that created them. (They are in svg, because of this it counts as lines of code)

**Review by dsrparracho — APPROVED** [review]

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 1]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+### 4. Data Negotiation and Transfer Phase
+
+Once we have the submodel we are going to call the [`subprotocolBody`](#L233) url of the `endpoint interface` with name `SUBMODEL-3.0`. This will provide for us the asset id to negotiate with the EDC Provider. Once this asset is negotiated we will request for the `transfer` and `EDR` token will be sent to the backend by the EDC Provider, allowing us to query the dataplane url contained in the `href` field of the endpoint interface. And in this way we will retrieve the data using the `EDC Provider Data Plane Proxy`.
+
+
+## Technical Detailed Phases
+
+## 1. Discovery Phase + 2. Digital Twin Registry Search Phase
+
+At the discovery phase the digital product pass application (Eco Pass KIT) will search in the Discovery Service for the following keys:
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:96]: "EcoPass"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 2]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+
+| Key                | Service Name          |
+| ------------------ | --------------------- |
+| bpn                | EDC Discovery Service |
+| manufacturerPartId | BPN Discovery Service |
+
+By searching for this keys the application will find the necessary services and will be able to search for the information needed to start the search for digital twin registries.
+
+After the discovery phase, the search for digital twin registries is one of the core phases to be done when retrieving data in Catena-X. Is necessary for filtering the EDCs which contain the digital twin registry assets, allowing an optimized search without high waiting times.
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:105]: "**It** is necessary"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 3]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+
+### Prerequisites
+
+The following information is required to enable the decentralized search for digital twin registries:
+
+| Name            | Example              | Description                                                                                                                                                                                                                                                                                       |
+| --------------- | -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Search Id Type  | *manufacturerPartId* | The search id type is required first of all to know in which `BPN Discovery` services to search. It will be introduced in the `Discovery Service` and we will obtain a list of `BPN Discovery Endpoints`. After this same id will be introduce as the *`type`* attribute in each `BPN Discovery`. |
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:115]: "After this same id will be introduce**d** as the type [...]"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 4]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+| Search Id Value | *XYZ78901*           | The search id value is required for searching in the `BPN Discovery` services. One example could be the `product type id` of a company, which is owned by an unique `BPN` reducing the complexity of the search.                                                                                  |
+
+### Sequence Diagram
+
+This sequence diagram represents the digital twin search and the discovery phases. For more information [go to explanation](#2-digital-twin-registry-search-phase)
+[![Sequence Diagram](./resources/development-view/developmentview-sequence-diagramm.svg)](./resources/development-view/developmentview-sequence-diagramm.svg)
+
+> **NOTE**: For learning how to register the assets and the digital twin registry and operate the Eco Pass KIT visit the [Operation View](./page-software-operation-view.md)
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:123]: "EcoPass"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 5]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+| ---------------------- | ---------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Specific Asset Id Type | *partInstanceId* | The specific asset id type is used to search in the `digital twin registry` for an specific digital twin. It is basically the `name` of  "specificAssetId" object located at the [`digital twin`](#aas-30-digital-twin-example) `specificAssetIds` property. The `*partInstanceId*` is used as an example most of the time, since the digital twin registry implemented a hotfix that allows companies say who can access to their `partInstanceId` fields. Now allowing the *"PUBLIC_READABLE"* property. |
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:136]: "for **a** specific digital twin"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 6]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+[![Negotiation and Transfer](./resources/development-view/ecoPassNegotiationAndTransferSequence.svg)](./resources/development-view/ecoPassNegotiationAndTransferSequence.svg)
+
+
+## Authentication and Authorization
+
+The authentication is administrated by the `IAM Services` from the Portal defined in the CX - 0001 Standard
+
+The backend APIs are authenticated by using a `JWT Token` in the request as `Bearer` token. The frontend is responsible for obtaining this token in the portal federated IAM.
+
+The Eco Pass KIT has two authorization methods:
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:165]: "EcoPass"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 7]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+| Portal Roles                  | Each digital product pass application contains an specific "AppId" provided by the portal in the Marketplace registration. And this Id shall be added to the configuration of the DPP Application in order to authenticate the user. If the end user contains any role added in the portal it will have access to the application if enabled. |
+| Business Partner Number (BPN) | The digital product pass application contains a check for the "BPN" of the end user. An option is also to allow the user to login if he is belonging to the company of the configured "EDC" so no user can act in name of a specific company.                                                                                                 |
+
+> **NOTE**: The authorization can be configure in the Eco Pass KIT configuration so that it matches the business interests of the Operator.
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:172]: "configure**d**"
"EcoPass"
"**o**perator"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 8]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
-| EDC Discovery Service | A microservice that resolves a BPN against an EDC endpoint.                            | [Tractus-X Portal including EDC Discovoery API](https://github.com/eclipse-tractusx/portal-backend) | [CX - 0001](https://catena-x.net/fileadmin/user_upload/Standard-Bibliothek/Update_PDF_Maerz/9_Data-Discovery-Services/CX_-_0001_EDC_DISCOVERY_API_PlatformCapabilityDS_v_1.0.1-1.pdf)    |
-| EDC                   | Eeclipse Dataspace Connector                                                           | [Tractus-X EDC](https://github.com/eclipse-tractusx/tractusx-edc)                                   | [CX - 0018](https://catena-x.net/fileadmin/user_upload/Standard-Bibliothek/Update_PDF_Maerz/3_Sovereign_Data_Exchange/CX_-_0018_EDC_PlatformCapabilitySovereignDataExchange_v_1.0.1.pdf) |
+### Open API specification
+
+The Digital Product Pass / Eco Pass KIT Open API specification is available at the swagger hub from Tractus-X:
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:182]: "EcoPass"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 9]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+
+[https://app.swaggerhub.com/apis/eclipse-tractusx-bot/digital-product-pass](https://app.swaggerhub.com/apis/eclipse-tractusx-bot/digital-product-pass)
+
+### EcoPass APIs
+The APIs below are the ones contain in the `Digital Product Pass Backend` reference implementation. Which can be reused for retrieving aspects from the Catena-X Network.
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:187]: "contain**ed**" ?

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 10]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+ | **/api/contract/create**              | POST   | The `/api/contract/create` api is responsible for calling the `BPN Discovery` service searching for the BPN of a `manufacturerPartId` and validating if there is any `Decentral Digital Twin Registry` available for the BPN number found in the `EDC Discovery` service.                                                                                                      | [Go to Params](#apicontractcreate)
+ | **/api/contract/search**              | POST   | At the **/api/contract/search**  API the user can search for a serialized Id and get its contract. The `Backend` will search for the Digital Twin and will return the contract for the first one that is found. A `sign token` (a sha256 hash) is return also and acts like a "session token" allowing just the user that created the process to sign or decline the contract. |[Go to Params](#apicontractsearch) |
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:192]: "A sign token is return**ed** also and acts ...."

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 11]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+ | **/api/contract/status/`<processId>`** | GET    | After the user signs the contract he can use the `/status` API to get the process status and see when it is ready to retrieve the passport using the API `/data`..                                                                                                                                                                                                             |         [Go to Params](#apicontractstatusprocessid)                                                                                                                                                                                                         
+ | **/api/data**                         | POST   | The API `/data` will decrypt the passport file that is encrypted using the session token "sign token", and will delete the file so that it is returned just once to the user and can not be accessed anymore. So a new passport will be always need to be requested..                                                                                                          |  [Go to Params](#apidata)  |
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:197]: Just one dot should be enough in "so a new passport will be always need to be requested"

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 12]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+ | **/api/contract/cancel**              | POST   | The user can use `/cancel` to interrupt the negotiation process once it is signed by mistake if is the case. It will be only valid until the negotiation is made.                                                                                                                                                                                                              |    [Go to Params](#apicontractcancel)                                                                                                                           
+ | **/api/contract/status/`<processId>`** | GET    | After the user signs the contract he can use the `/status` API to get the process status and see when it is ready to retrieve the passport using the API `/data`..                                                                                                                                                                                                             |         [Go to Params](#apicontractstatusprocessid)
```
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:196]: One dot after "using the API data" should be sufficient

File: docs-kits/kits/Eco_Pass_KIT/resources/development-view/ecoPassSearchSequence.svg
Comment (maxim-bo) [inline docs-kits/kits/Eco_Pass_KIT/resources/development-view/ecoPassSearchSequence.svg:1]: "Tran**s**fer completed" (it is in the /search process after the transfer is triggered)

**Review by maxim-bo — CHANGES_REQUESTED** [review]
Some minor typos and grammar issues found, apart from that no complaints!

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 13]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+## Data Retrieval Flow
+In order to achieve a better understanding of the Eco Pass KIT data retrieval flow, we can detail a specific example where an user wants to retrieve an specific passport for a asset in Catena-X using the Eco Pass KIT (reference implementation [`digital-product-pass`](https://github.com/eclipse-tractusx/digital-product-pass)).
```
[the remark this answers is not in the document]
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:23]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 14]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+CX:<manufacturerPartId>:<partInstanceId>
+
+```
+
+
+
+| ID                     | Description                                                                                                                                                                                              |
+| ---------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `CX`                   | The "**CX**" argument is the one that indicates that this ID is a Catena-X Search Identificator.                                                                                                          |
+| `<manufacturerPartId>` | The "**manufacturerPartId**" makes reference to the id type of the product. Is the identification the manufacturer gives to an specific part he produced.                                                |
```
[the remark this answers is not in the document]
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:46]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 15]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+| EDC                   | Main gateaway to the network. In this use case two edc need be existing, one connected to the Digital Product Pass (EcoPass KIT) [EDC Consumer] and another to the Provider Catena-X components [EDC Provider]         | [eclipse-tractusx/tractusx-edc](https://github.com/eclipse-tractusx/tractusx-edc)                                                                                                                                                                                                                            | CX - 0018                                                   |
```
[the remark this answers is not in the document]
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:62]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 1 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:96]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 2 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:105]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 3 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:115]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 4 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:123]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 5 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:136]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 6 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:165]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 7 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:172]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 8 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:182]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 9 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:187]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 12 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:196]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 11 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:197]: ok, weird why it happens

File: docs-kits/kits/Eco_Pass_KIT/resources/development-view/ecoPassSearchSequence.svg
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/resources/development-view/ecoPassSearchSequence.svg:1]: done

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 16]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+
+| Parameter | Value Name            | Mandatory or Optional Value |
+|-----------|-----------------------|-----------------------------|
+| id        | serializedIdValue     | [REQUIRED]                  |
+| idType    | serializedIdTypeName  | partInstanceId              |
+| processId | processIdentification | [REQUIRED]                  |
 
-Please note: In order to provide a passport asset for consumption a corresponding offer must exist. Please refer to “[Setup Data Offer](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/docs/samples/Transfer%20Data.md#2-setup-data-offer)” in the EDC documentation.
 
-## Reference Implementationfernce
```
Comment (maximilianong) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:50]: Typo Implementation**fernce**

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md [hunk 17]
```diff
@@ -11,43 +13,263 @@ Thereby, this KIT covers various aspects, starting from how the available API En
...
+| Service   | Method | Path                                               | Params                                     |
+| ------ | ------ | -------------------------------------------------- | ------------------------------------------ |
+| Discovery Service [CX-0053] | POST   | /api/administration/connectors/discovery/search    | key:manufacturerPartId (BPN Discovery) & key:bpn (EDC Discovery)                          |
+| BPN Discovery Service [CX-0053]  | POST   | /api/administration/connectors/bpnDiscovery/search | key:manufacturerPartId                          |
+| EDC Discovery Service [CX-0001]  | POST   | /api/administration/connectors/discovery/search    | `{[<Company's BPNL>]}`                     |
+| Digital Twin Registry [CX-0002]  | GET    | /lookup/shells                                     | key:partInstanceID, value: batteryDMC-Code or Product Identification Code |
+| Digital Twin Registry [CX-0002]  | GET    | /shell-descriptors/{digitalTwinId}                                     | key:partInstanceID, value: batteryDMC-Code or Product Identification Code |
+
+
+## Reference Implementations
```
Comment (maximilianong) [inline docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:272]: If you have an API as reference implementation please add this to the KIT via OpenAPI Plugin e.g. : https://eclipse-tractusx.github.io/docs-kits/kits/Business%20Partner%20Kit/Software%20Development%20View/Pool%20Api/get-sites-paginated

**Review by maximilianong — CHANGES_REQUESTED** [review]

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 16 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:50]: oh good that you saw that :)

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 17 above)
Comment (matbmoser) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:272]: The API plugin will no be anymore supported.

We have referenced our OpenApi specification here: 

image

https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/631/files/2679879bf3729d525fb9d43a660e51921b0aa823#diff-c8b6ae86bef984ee00e1927ccc4407795adcb2b5757dd0c799d71a23d160bd99R184

In this matrix chat you can confirm my afirmation: https://chat.eclipse.org/#/room/!oXNwXGsvkbDUMiQhms:matrix.eclipse.org/$W63RrxowCqp9qmwF4QkfcjumeyhZ1dWIo2udbbrWyAk?via=matrix.eclipse.org&via=matrix.org&via=beeper.com


It was stated by @SebastianBezold here: https://chat.eclipse.org/#/room/!oXNwXGsvkbDUMiQhms:matrix.eclipse.org/$jRRigM3k-kVYdLhY0K4eHfPFhDqFaOLDKVtRQedUQyY?via=matrix.eclipse.org&via=matrix.org&via=beeper.com

**Comment by matbmoser** [comment]
Linting problems are not from EcoPass KIT

File: docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md (hunk 17 above)
Comment (maximilianong) [reply docs-kits/kits/Eco_Pass_KIT/page-software-development-view.md:253]: We are still maintaining the plugin but if you are using Swaggerhub, better, thanks.

**Review by danielmiehle — APPROVED** [review]

**Comment by maximilianong** [comment]
@matbmoser please resolve the conflict we need to merge it today for the release.

**Comment by matbmoser** [comment]
@maximilianong done!

**Review by saudkhan116 — APPROVED** [review]

**Review by LuisRickert — APPROVED** [review]

**Review by maximilianong — APPROVED** [review]

**Pull request merged**